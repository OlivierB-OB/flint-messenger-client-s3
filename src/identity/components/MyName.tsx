import Typography from '@material-ui/core/Typography';
import React from 'react';
import { connect } from 'react-redux';
import { IAppState } from '../../appReducer';
import { UserName, IUserNameProps } from '../../users/components/UserName';

type IMyNameDisplayProps = Partial<IUserNameProps>;

function MyNameDisplay({ info }: IMyNameDisplayProps) {
  if (!info) return null;
  return (
    <Typography variant="h6">
      <UserName info={info} />
    </Typography>
  );
}

const mapStateToProps = ({ identity }: IAppState) => ({
  info: identity.info,
});

export const MyName = connect(mapStateToProps)(MyNameDisplay);
