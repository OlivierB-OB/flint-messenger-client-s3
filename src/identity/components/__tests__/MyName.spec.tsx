import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import { MyName } from '../MyName';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('MyName', () => {
  let Connected: any = MyName;

  it('should provide user info to MyNameDisplay', async () => {
    const store = mockStore({
      identity: {
        info: 'the user information',
      },
    });
    const component = shallow(<Connected store={store} />).dive();
    expect(component).toMatchSnapshot();
  });
});
