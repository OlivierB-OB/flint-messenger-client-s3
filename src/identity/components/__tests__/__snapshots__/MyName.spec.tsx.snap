// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyName should provide user info to MyNameDisplay 1`] = `
<MyNameDisplay
  dispatch={[Function]}
  info="the user information"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
